impl/client/src/main/doc-js/pentaho/type/spec/IInstanceResolve.jsdoc

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright 2017 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `spec.IInstanceResolve` interface describes a special syntax used
+ * for resolving instances from a Type API [instance container]{@link pentaho.type.InstancesContainer}.
+ *
+ * ##### Resolve instance by id syntax
+ *
+ * Denotes the named instance with the given id. The instance must be registered.
+ *
+ * ```json
+ * {$instance: {id: "my/service"}}
+ * ```
+ *
+ * The properties
+ * [isRequired]{@link pentaho.type.spec.IInstanceResolve#isRequired} and
+ * [reservation]{@link pentaho.type.spec.IInstanceResolve#reservation},
+ * can also be specified.
+ *
+ * ##### Resolve instance by element type syntax
+ *
+ * Denotes the highest priority registered instance of a given [element]{@link pentaho.type.Element} type.
+ *
+ * ```json
+ * {$instance: {type: "my/Service"}}
+ * ```
+ *
+ * The properties
+ * [filter]{@link pentaho.type.spec.IInstanceResolve#filter},
+ * [isRequired]{@link pentaho.type.spec.IInstanceResolve#isRequired} and
+ * [reservation]{@link pentaho.type.spec.IInstanceResolve#reservation},
+ * can also be specified.
+ *
+ * ##### Resolve instance by list type syntax
+ *
+ * Denotes the list of all of the registered instances of
+ * the given list type's [element type]{@link pentaho.type.List.Type#of},
+ * ordered from highest to lowest [priority]{@link pentaho.type.spec.IInstanceDeclaration#priority}.
+ *
+ * ```json
+ * {$instance: {type: ["my/Service"]}}
```

Comment:
Like I mentioned I would prefer to have an alternative syntax to represent getting the list of all registered instances of a given type. For later discussion on possible alternatives or if the current representation is the one we want to stick with.